[HttpClient] Fix retrying requests with `Psr18Client` and NTLM connections

nicolas-grekas opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 7.2
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | Fix #59489, Fix #59629
| License       | MIT

nicolas-grekas committed:
[HttpClient] Fix retrying requests with Psr18Client and NTLM connections

veewee commented on src/Symfony/Component/HttpClient/Psr18Client.php:
```diff
@@ -98,13 +98,26 @@ public function sendRequest(RequestInterface $request): ResponseInterface
             $options = [
                 'headers' => $headers,
-                'body' => static fn (int $size) => $body->read($size),
+                'body' => 0 <= $size && $size < 1 << 21 ? $body->getContents() : static function (int $size) use ($body) {
```
Can this logic also be applied to `HttplugClient`?

nicolas-grekas replied:
sure, updated !

ajgarlag committed:
Prevent empty request body stream in HttplugClient and Psr18Client

This prevents adding `Content-Length` and `Transfer-Encoding` headers
when sending a request with an empty body using CurlHttpClient

nicolas-grekas commented:
Thank you @ajgarlag.
